### PR TITLE
Follow up of PSD2 updates

### PR DESCRIPTION
**Pull request by valoricDe**
It is the addition taken from Guido Müller and reverted some stuff like the badges.
Only thing left I guess is the stuff with the e2e test

### CODE REVIEW

File: packages/fints/README_advanced_usage.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            getAccountStatements( client, accounts, statements, callback )
+          }
+        } );
+      break
+    }
+  }
+}
+
+
+const client = new PinTanClient( {
```
Comment (Prior99) [inline packages/fints/README_advanced_usage.md:30]: Please fix alignment here.

File: packages/fints/package.json [hunk 2]
```diff
@@ -1,12 +1,15 @@
 {
     "name": "fints",
-    "version": "0.3.1",
-    "description": "FinTS client library.",
+    "version": "0.5.1",
```
Comment (Prior99) [inline packages/fints/package.json:3]: Version 0.4.0 should be sufficient.

File: packages/fints/src/__tests__/test-pin-tan-client-e2e.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import { SEPAAccount } from "../types";
+
+const url = process.env.FINTS_URL;
+const name = process.env.FINTS_USER;
+const pin = process.env.FINTS_PASSWORD;
+const blz = process.env.FINTS_BLZ;
+
+const productId = "9FA6681DEC0CF3046BFC2F8A6";
+
+test.skip("get accounts", async () => {
```
Comment (Prior99) [inline packages/fints/src/__tests__/test-pin-tan-client-e2e.ts:13]: I suggest to remove the `.skip` and add another test command for jest that only tests the e2e files. Also I would appreciate some documentation on how exactly to use these tests and what for. Or simply delete this file.

File: packages/fints/src/types.ts [hunk 4]
```diff
@@ -150,6 +152,24 @@ export interface SEPAAccount {
      * The account's BLZ used together with the account number before SEPA.
      */
     blz: string;
+
+    accountOwnerName?: string;
```
Comment (Prior99) [inline packages/fints/src/types.ts:156]: Would be awesome if you could add some documentation here.

File: packages/fints/jest.config.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (Prior99) [inline packages/fints/jest.config.json:1]: Why was this file added but the configuration for jest not removed in the package.json? Either delete the file or the section in the package.json please :)

**Review by Prior99 — COMMENTED** [review]
Thanks for the changes. I have only a few comments that should be easy to fix.
Please contact me if you need help.

I have the problem that PSD2 basically made FinTS useless for me as I can no longer automatically and unattendendly access my banking data now.

Also I have no PSD2 key and no need for one, hence I can not test this.

Can anybody recommend a simple PSD2 Sandbox test server that I can use to test this?

**Comment by valoricDe** [comment]
When I run the unit tests I get: 
```
 FAIL  src/__tests__/test-pin-tan-client.ts (12.412s)
  ● accounts

    TypeError: fetchMock.post is not a function

      24 |     const responseFixtures: string[] = JSON.parse(readFileSync(`${__dirname}/fixture-accounts.json`, "utf8"));
      25 |     let responseNo = 0;
    > 26 |     const mock = fetchMock.post(url, () => {
         |                            ^
      27 |         const response = encodeBase64(responseFixtures[responseNo]);
```
do you have an idea why?

File: packages/fints/src/types.ts (hunk 4 above)
Comment (valoricDe) [reply packages/fints/src/types.ts:156]: I found https://www.hbci-zka.de/dokumente/spezifikation_deutsch/fintsv3/FinTS_3.0_Messages_Geschaeftsvorfaelle_2015-08-07_final_version.pdf but I don't know how to read it. If you know maybe we can have a zoom call. Afterwards I would be happy to add it here.

**Comment by Prior99** [comment]
I will have a look at this the next days, when I can find the time. Thanks :)

**Comment by Prior99** [comment]
I took a look at this. The lining still fails, `fints-cli` depends on `fints` in version `0.5.1`, but `fints` currently is at version `0.4.0` according to the `package.json`. Also, these E2E tests are still not executable, or am I mistaken?

**Comment by valoricDe** [comment]
Good catch.
For the e2e tests: Did you provide your bank account information as env variables?

**Comment by valoricDe** [comment]
FYI, DKB now also requires a tan when fetching the account data. This library could probably act on it. But I guess this might involve further work

**Comment by Prior99** [comment]
Thanks so much for you contribution. And sorry for the horrendous delay from my site. I hope everything works well now. If you ever need professional help with this package, feel free to contact me, I am always looking for interesting projects.

**Comment by Prior99** [comment]
Also, @valoricDe and @mathiasarens  please feel free to create a new PR adding yourself as contributors to the README.
